tests/integration_tests/functional/test_api.py

```diff
@@ -940,52 +945,76 @@ def test_api_version(test_microvm_with_api):
     assert out.strip()[1:] == preboot_response.json()['firecracker_version']
 
 
-def test_api_vsock(test_microvm_with_api):
+def test_api_vsock(bin_cloner_path):
     """
     Test vsock related API commands.
 
     @type: functional
     """
-    test_microvm = test_microvm_with_api
-    test_microvm.spawn()
-    test_microvm.basic_config()
+    builder = MicrovmBuilder(bin_cloner_path)
+    artifacts = ArtifactCollection(_test_images_s3_bucket())
+    # Fetch 0.25 and older firecracker binaries.
+    # Create a vsock device with each FC binary
+    # artifact.
+    firecracker_artifacts = artifacts.firecrackers(
+        # v1.0.0 deprecated `vsock_id`.
+        min_version="1.0.0")
```

Comment:
A couple of things:

1. Should we check if `vsock_id` is deprecated only if the artifact version < `2.0.0` ? Right now `artifacts.firecrackers()` has a parameter `max_version`, but it wouldn't be a perfect fit, because it would check if `version <= max_version`.

2. Also should we check if `vsock_id` is deprecated only if the version of the current firecracker build is `>= 1.0.0` and `< 2.0.0` ? I don't think it makes sense to do this check if current firecracker version is `9.0.0` for example .

3. Apart from the artifacts, should we also check the current firecracker build ? Otherwise we may introduce a bug in the current build and the tests can pass, because the artifacts are working.

Reply:
We are supposed to check with future binaries from this major version only. This will be useful only when running tests on older supported branches for things like patch releases.

I added a test that runs with the current build as well.